common-api/src/main/scala/pl/touk/nussknacker/engine/api/TypeSpecificData.scala

```diff
@@ -0,0 +1,133 @@
+import pl.touk.nussknacker.engine.api.TypeSpecificUtils.{convertPropertyWithLog, toStringMap}
+
+@ConfiguredJsonCodec sealed trait TypeSpecificData {
+  val isSubprocess = this match {
+    case _: ScenarioSpecificData => false
+    case _: FragmentSpecificData => true
+  }
+  val toProperties: Map[String, String]
+}
+
+object TypeSpecificData {
+  def apply(properties: Map[String, String], propertiesType: String): TypeSpecificData = {
+    propertiesType match {
+      case "StreamMetaData" => StreamMetaData(properties)
+      case "LiteStreamMetaData" => LiteStreamMetaData(properties)
+      case "RequestResponseMetaData" => RequestResponseMetaData(properties)
+      case "FragmentSpecificData" => FragmentSpecificData(properties)
+      case _ => throw new IllegalStateException(s"Unrecognized scenario metadata type: ${propertiesType}")
+    }
+  }
+}
+
+case class FragmentSpecificData(docsUrl: Option[String] = None) extends TypeSpecificData {
+  override val toProperties: Map[String, String] = toStringMap(List(
+    docsUrl.map(docsUrlName -> _)
+  ))
+}
+
+object FragmentSpecificData {
+  private val docsUrlName = "docsUrl"
+
+  def apply(properties: Map[String, String]): FragmentSpecificData = {
+    FragmentSpecificData(
+      docsUrl = properties.get(docsUrlName)
+    )
+  }
+}
+
+sealed trait ScenarioSpecificData extends TypeSpecificData
+
+// TODO: rename to FlinkStreamMetaData
+case class StreamMetaData(parallelism: Option[Int] = None,
+                          //we assume it's safer to spill state to disk and fix performance than to fix heap problems...
+                          spillStateToDisk: Option[Boolean] = Some(true),
+                          useAsyncInterpretation: Option[Boolean] = None,
+                          checkpointIntervalInSeconds: Option[Long] = None) extends ScenarioSpecificData {
+
+  def checkpointIntervalDuration: Option[Duration] = checkpointIntervalInSeconds.map(Duration.apply(_, TimeUnit.SECONDS))
+
+  override val toProperties: Map[String, String] = toStringMap(List(
+    parallelism.map(parallelismFlinkName -> _),
+    spillStateToDisk.map(spillStateToDiskName -> _),
+    useAsyncInterpretation.map(useAsyncInterpretationName -> _),
+    checkpointIntervalInSeconds.map(checkpointIntervalInSecondsName -> _)
+  ))
+}
+
+object StreamMetaData {
+  private val parallelismFlinkName = "parallelism"
+  private val spillStateToDiskName = "spillStateToDisk"
+  private val useAsyncInterpretationName = "useAsyncInterpretation"
+  private val checkpointIntervalInSecondsName = "checkpointIntervalInSeconds"
+
+  def apply(properties: Map[String, String]): StreamMetaData = {
+    StreamMetaData(
+      parallelism = properties.get(parallelismFlinkName).flatMap(convertPropertyWithLog(_, _.toInt, parallelismFlinkName)),
+      spillStateToDisk = properties.get(spillStateToDiskName).flatMap(convertPropertyWithLog(_, _.toBoolean, spillStateToDiskName)),
```

Comment:
let's have fallback to `Some(true)` here

Reply:
Thanks, working on it now